terraform-beyond-layer0/example-2/main.tf

```diff
@@ -0,0 +1,91 @@
+#
+# ./example2/main.tf
+#
+
+provider "layer0" {
+  endpoint        = "${var.endpoint}"
+  token           = "${var.token}"
+  skip_ssl_verify = true
+}
+
+provider "aws" {
+  access_key = "${var.access_key}"
+  secret_key = "${var.secret_key}"
+  region     = "${var.region}"
+}
+
+# create one layer0 environment for each map value defined in variable 'environments'
+resource "layer0_environment" "demo" {
+  count = "${length(var.environments)}"
+
+  name = "${lookup(var.environments, count.index)}_demo"
+}
+
+# create one loadbalancer for each map value defined in variable 'environments'
+resource "layer0_load_balancer" "guestbook" {
+  count = "${length(var.environments)}"
+
+  name        = "${lookup(var.environments, count.index)}_guestbook_lb"
+  environment = "${element(layer0_environment.demo.*.id, count.index)}"
+
+  port {
+    host_port      = 80
+    container_port = 80
+    protocol       = "http"
+  }
+}
+
+# create a datasource for each map value defined in variable 'environments' which
```

Comment:
[nit] I had to re-read this a couple time to make sense of it. Perhaps we should "dumb it down":
```
Create a template for each environment. Each template will reference an environment-specific dynamodb table. 
```